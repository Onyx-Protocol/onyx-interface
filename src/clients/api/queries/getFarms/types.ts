import BigNumber from 'bignumber.js';
import { Multicall } from 'ethereum-multicall';
import { EthChainId, Token } from 'types';

export interface Address {
  [EthChainId.MAINNET]: string;
  [EthChainId.TESTNET]: string;
}

export interface FarmConfig {
  pid: number;
  lpSymbol: Uppercase<string>;
  lpAddresses: Address;
  token: Token;
  quoteToken: Token;
  multiplier?: string;
}

export interface Farm extends FarmConfig {
  tokenAmount?: BigNumber;
  quoteTokenAmount?: BigNumber;
  lpTotalInQuoteToken?: BigNumber;
  tokenPriceVsQuote?: BigNumber;
  poolWeight?: BigNumber;
  lpTokenBalanceMC?: BigNumber;
  tokenPerSecond?: BigNumber;
  farmApr?: number;
  userData?: {
    allowance: BigNumber;
    tokenBalance: BigNumber;
    stakedBalance: BigNumber;
    earnings: BigNumber;
  };
}

export interface GetFarmsInput {
  multicall: Multicall;
  accountAddress: string;
}

export type GetFarmsOutput = {
  farms: Farm[];
};

export const getAddress = (address: Address): string => {
  const chainId: EthChainId = process.env.REACT_APP_CHAIN_ID
    ? parseInt(process.env.REACT_APP_CHAIN_ID, 10)
    : EthChainId.MAINNET;
  return address[chainId];
};
