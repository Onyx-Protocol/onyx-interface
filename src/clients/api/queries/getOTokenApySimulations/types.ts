import BigNumber from 'bignumber.js';
import { Multicall } from 'ethereum-multicall';

export interface GetOTokenInterestRatesInput {
  multicall: Multicall;
  reserveFactorMantissa: BigNumber;
  interestRateModelContractAddress: string;
}

export interface OTokenApySnapshot {
  utilizationRate: number;
  borrowApyPercentage: number;
  supplyApyPercentage: number;
}

export type GetOTokenApySimulationsOutput = {
  apySimulations: OTokenApySnapshot[];
};
