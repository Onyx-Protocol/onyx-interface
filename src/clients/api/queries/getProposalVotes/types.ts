import { ProposalApiResponse } from '../getProposals';

export interface GetProposalVotesInput {
  address?: string;
}

export type ProposalVotesApiResponse = {
  id: string;
  proposal: ProposalApiResponse;
  address: string;
  support: boolean;
  votes: string;
  blockNumber: string;
  blockTimestamp: string;
  transactionHash: string;
};

export interface ProposalVotesDetailApiResponse {
  proposalsVotes: ProposalVotesApiResponse;
}

export interface GetProposalVotesOutput {
  total: number;
  proposalVotes: ProposalVotesApiResponse[];
}
