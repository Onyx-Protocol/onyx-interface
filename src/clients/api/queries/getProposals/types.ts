import { Proposal } from 'types';

export interface GetProposalsInput {
  limit?: number;
  page?: number;
}

export interface GetProposalInput {
  id: number | string;
}

export interface ProposalApiResponse {
  call_datas?: string | null;
  callDatas?: string | null;
  signatures: string;
  targets: string;
  values: string;
  against_votes?: string | null;
  againstVotes?: string | null;
  cancel_block?: number | null;
  cancel_timestamp?: number | null;
  cancel_tx_hash?: string | null;
  cancelBlock?: number | null;
  cancelTimestamp?: number | null;
  cancelTxHash?: string | null;
  canceled: number | boolean;
  created_at?: string | null;
  createdAt?: string | null;
  created_block: number | null;
  created_timestamp: number | null;
  created_tx_hash?: string | null;
  description: string;
  end_block?: number;
  end_timestamp?: number | null;
  end_tx_hash?: string | null;
  endBlock?: number;
  endTimestamp?: number | null;
  endTxHash?: string | null;
  eta: number;
  executed: number | boolean;
  executed_block: number | null;
  executed_timestamp: number | null;
  executed_tx_hash: string | null;
  for_votes?: string | null;
  forVotes?: string | null;
  id: number;
  proposer: string;
  queued_block: number | null;
  queued_timestamp: number | null;
  queued_tx_hash: string | null;
  start_block?: number | null;
  start_timestamp?: number;
  start_tx_hash?: string | null;
  startBlock?: number | null;
  startTimestamp?: number;
  startTxHash?: string | null;
  state:
    | 'Pending'
    | 'Active'
    | 'Canceled'
    | 'Defeated'
    | 'Successded'
    | 'Queued'
    | 'Expired'
    | 'Executed';
  // JSON
  updated_at?: string | null;
  updatedAt?: string | null;
  voter_count?: number | null;
  voterCount?: number | null;
}

export interface ProposalsApiResponse {
  data: ProposalApiResponse[];
  metadata: {
    page: number;
    limit: number;
    totalItem: number;
    totalPage: number;
  };
}

export interface ProposalDetailApiResponse {
  data: ProposalApiResponse;
  metadata: {
    page: number;
  };
}

export interface GetProposalsOutput {
  page: number;
  limit: number;
  total: number;
  proposals: Proposal[];
}

export type GetProposalOutput = Proposal;
