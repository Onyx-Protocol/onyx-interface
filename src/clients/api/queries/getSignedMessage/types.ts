export interface GetSignedMessageResponse {
  nonce: string;
  timestamp: number;
}
