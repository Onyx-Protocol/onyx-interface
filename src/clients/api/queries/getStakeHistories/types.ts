import { StakeHistory } from 'types';

export interface GetStakeHistoriesInput {
  limit?: number;
  page?: number;
  address?: string;
}

export interface StakeHistoryApiResponse {
  id: number;
  address: string;
  tx_hash: string;
  block_hash: string;
  amount: string;
  price: string;
  reward: string;
  type: number;
  created_at: string;
  updated_at: string;
}

export interface StakeHistoriesApiResponse {
  data: StakeHistoryApiResponse[];
  metadata: {
    page: number;
    limit: number;
    totalItem: number;
    totalPage: number;
  };
}

export interface GetStakeHistoriesOutput {
  page: number;
  limit: number;
  total: number;
  stakeHistories: StakeHistory[];
}
