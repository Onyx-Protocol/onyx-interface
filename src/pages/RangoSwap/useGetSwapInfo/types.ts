import { Trade as PSTrade } from '@uniswap/sdk';
import { Swap, SwapDirection, Token } from 'types';

export type SwapError = 'INSUFFICIENT_LIQUIDITY' | 'WRAPPING_UNWRAPPING_UNSUPPORTED';

export interface UseGetSwapInfoInput {
  fromToken: Token;
  toToken: Token;
  direction: SwapDirection;
  fromTokenAmountTokens?: string;
  toTokenAmountTokens?: string;
}

export interface UseGetSwapInfoOutput {
  swap: Swap | undefined;
  error: SwapError | undefined;
}

export interface FormatToSwapInput {
  input: UseGetSwapInfoInput;
  trade: PSTrade;
}

export type FormatToSwapOutput = Swap;
