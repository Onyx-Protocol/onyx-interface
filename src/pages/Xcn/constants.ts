import BigNumber from 'bignumber.js';

import { TOKENS } from 'constants/tokens';

export const MINTED_XCN = '23700000';
export const MINTED_XCN_WEI = new BigNumber(MINTED_XCN).times(
  new BigNumber(10).pow(TOKENS.xcn.decimals),
);
