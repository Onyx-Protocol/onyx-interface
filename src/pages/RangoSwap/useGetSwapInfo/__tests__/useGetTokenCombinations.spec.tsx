import React from 'react';
import { PSTokenCombination } from 'types';

import { UNISWAP_TOKENS } from 'constants/tokens';
import renderComponent from 'testUtils/renderComponent';

import useGetTokenCombinations from '../useGetTokenCombinations';

describe('pages/Swap/useGetSwapInfo/useGetTokenCombinations', () => {
  it('returns all possible combinations between the tokens provided and the base trade ones', () => {
    let tokenCombinations: PSTokenCombination[] = [];

    const TestComponent = () => {
      tokenCombinations = useGetTokenCombinations({
        fromToken: UNISWAP_TOKENS.usdt,
        toToken: UNISWAP_TOKENS.uni,
      });

      return <></>;
    };

    renderComponent(TestComponent);

    // Test there's no duplicates
    let hasDuplicate = false;

    tokenCombinations.reduce((acc, tokenCombination) => {
      hasDuplicate = acc.some(
        prevTokenCombination =>
          prevTokenCombination[0].address === tokenCombination[0].address &&
          prevTokenCombination[1].address === tokenCombination[1].address,
      );

      return [...acc, tokenCombination];
    }, [] as PSTokenCombination[]);

    expect(hasDuplicate).toBe(false);

    expect(tokenCombinations).toMatchSnapshot();
  });

  it('uses wBNB when BNB is provided', () => {
    let tokenCombinations: PSTokenCombination[] = [];

    const TestComponent = () => {
      tokenCombinations = useGetTokenCombinations({
        fromToken: UNISWAP_TOKENS.eth,
        toToken: UNISWAP_TOKENS.uni,
      });

      return <></>;
    };

    renderComponent(TestComponent);

    expect(tokenCombinations).toMatchSnapshot();
  });
});
