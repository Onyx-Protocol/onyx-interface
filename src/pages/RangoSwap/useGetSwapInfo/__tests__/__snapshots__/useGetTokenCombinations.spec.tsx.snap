// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Swap/useGetSwapInfo/useGetTokenCombinations returns all possible combinations between the tokens provided and the base trade ones 1`] = `
Array [
  Array [
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 97,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "USDT",
    },
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "UNI",
    },
  ],
  Array [
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 97,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "USDT",
    },
    Token {
      "address": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "WETH",
    },
  ],
  Array [
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "UNI",
    },
    Token {
      "address": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "WETH",
    },
  ],
]
`;

exports[`pages/Swap/useGetSwapInfo/useGetTokenCombinations uses wBNB when BNB is provided 1`] = `
Array [
  Array [
    Token {
      "address": "0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6",
      "chainId": 5,
      "decimals": 18,
      "name": undefined,
      "symbol": "WETH",
    },
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 5,
      "decimals": 18,
      "name": undefined,
      "symbol": "UNI",
    },
  ],
  Array [
    Token {
      "address": "0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6",
      "chainId": 5,
      "decimals": 18,
      "name": undefined,
      "symbol": "WETH",
    },
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 5,
      "decimals": 6,
      "name": undefined,
      "symbol": "USDT",
    },
  ],
  Array [
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 5,
      "decimals": 18,
      "name": undefined,
      "symbol": "UNI",
    },
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 5,
      "decimals": 6,
      "name": undefined,
      "symbol": "USDT",
    },
  ],
]
`;

exports[`pages/Swap/useGetSwapInfo/useGetTokenCombinations uses wETH when BNB is provided 1`] = `
Array [
  Array [
    Token {
      "address": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "WETH",
    },
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "UNI",
    },
  ],
  Array [
    Token {
      "address": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "WETH",
    },
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 97,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "USDT",
    },
  ],
  Array [
    Token {
      "address": "0x651A2a9D3Fc81A5c95E8232d71a558FFd4111126",
      "chainId": 97,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "UNI",
    },
    Token {
      "address": "0xcE546eE9E8Bc0732A1e0083ef578a19F79c73D74",
      "chainId": 97,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": undefined,
      "projectLink": undefined,
      "symbol": "USDT",
    },
  ],
]
`;
